### DISCUSSION TITLE
Player screen design ideas

### DISCUSSION THREAD

**Discussion by M00NJ** [libre-tube/LibreTube#5247] [Show and tell]
I think the player screen could look a lot nicer so here's a mock-up with my ideas:

| Theme I | Theme II |
| --- | --- |
| ![Mockup](https://github.com/libre-tube/LibreTube/assets/82398591/caf74def-6a2e-4141-812b-f0dadbfc3574) | ![Mockup2](https://github.com/libre-tube/LibreTube/assets/82398591/2d2c0df3-3fd1-428e-ae46-500a7250df4c) |

**What's changed:**

1. Used individual buttons instead of a toolbar 
  (Looks more modern imo)
2. Removed the pop-up button
  (Because of the background playback issue you have to use `audio mode`, if you want to close the app and have the playback continue. That means having `PiP` disabled doesn't do anything since the playback will stop when you leave the app. So I suggest making `PiP` default and removing the pop-up button, to free up space in the player and settings screen.)
3. Replaced the comment expand button with the pinned comment
  (The space made available by replacing the toolbar can now be used to display important information, like the pinned comment. Clicking on it should still open the comment section.)
4. Increased the thumbnail size of the recommended videos
  (The new default video layout reduces the thumbnail size, which results in the titles being cramped or cut off oftentimes. Slightly increasing the thumbnail size and removing less important info like the upload date from the recommended section could fix this.)
5. Changes to button shapes, color mappings and padding

| **Screenshot** | **Mock-up** |
| --- | --- |
| ![LibreTube](https://github.com/libre-tube/LibreTube/assets/82398591/12e0b378-57c0-437c-a655-76a8ee504795) |  ![Mockup](https://github.com/libre-tube/LibreTube/assets/82398591/4d3cec31-199a-4074-82fa-3d797c85c06e) |

I used the same accent color in the comparison!

**Comment by Bnyro** [comment]
I really like these ideas:
I will implement 1, 2, 4 and 5 with the next release.
3. is unfortunately not really possible because loading the first comment would require an additional API request, which is not really a good idea in my opinion.

However, since we have a landscape layout for that page too now, we'll need to see how we change that to look good as well, or if we keep the current design there. That's because there is much more vertical free space because the related streams are placed somewhere else, which would cause a huge gap by removing the toolbar menu.

Apart from that I don't agree with the like/dislike icon design changes, the filled ones fit better imo.

**Comment by M00NJ** [comment]
**Amazing news!** 

Some info on the color mapping:
For the buttons I used `surface-container-low` for the background and `primary` for the text and icon (These are the same values as the `elevated button` but without the drop-shadow). The "subscribe button" is a default button. The background color I used is `surface` because imo the background in a streaming app should be darker. `surface-dim` is an option too, it keeps it's relative brightness. That means instead of being the brightest thing an screen in light mode, it will be a bit dimmer. 

I think commenting on the padding for the buttons won't be useful, because it'll probably have to relative to the screen size. 
I can have a look at the landscape layout and see how we can apply this design language there as well.
Here's a updated mock-up without the the pinned comment and filled thumb icons:

![Mockup](https://github.com/libre-tube/LibreTube/assets/82398591/162830c7-715d-4f48-8738-a72cb1fa324a)

**Reply by Bnyro** [reply]
I have already starting with implementing the new layout for the portrait mode (see https://github.com/libre-tube/LibreTube/pull/5301), you can already test the CI builds [here](https://github.com/libre-tube/LibreTube/actions/runs/7115922827). It's only a beginning and still needs some adjustments before the release, but it's already implementing the "major thing" so that only some smaller adjustments should be needed for the portrait mode now. Please feel free to test this already when you have time and give feedback about the differences you'd still like to see.

PS: I still want to decrease the size of the action buttons (like download, save, share, ...), but currently I can't seem to find a good way for that while using the builtin Android `MaterialButton`. Maybe we should use an elevated card instead in worst case.

**Comment by M00NJ** [comment]
I finally got around to testing it. This looks quite a bit nicer already :)
Here are some things apart from the smaller buttons, that would make it closer to the mock-up:

- Apply primary color to the comment button text and icon as well
- Choose a darker background color for the buttons (`surfaceContainerLow` /`surfaceColorAtElevation(1.dp)`)
- Choose a darker background color (`surface` or does it already use that by default? Looks so much darker in the muck-up...)

![signal-2023-12-15-102124](https://github.com/libre-tube/LibreTube/assets/82398591/8b133bd1-d17b-4463-8efa-b5adbaf470c5)

**Reply by Bnyro** [reply]
This should all be applied now.

Concerning the background color of the buttons: it somehow looks way darker when choosing a theme in the settings which is not "Material You", so I think we have to go with the current one and can't make it any darker.

**Reply by Bnyro** [reply]
The landscape layout is still completely missing though

**Comment by RafaelsRamos** [comment]
Hey @M00NJ 👋 I hope I'm not bothering you with this tag!

Would it be possible for you to give your input into the discussion on this PR #5366? Given that you provided the layouts in the first place, your opinion would be highly appreciated 💪 

Thanks in advance!

**Comment by M00NJ** [comment]
Since we can't change the width of the buttons in the row under the video I think we should discuss other options. I really like the new look and darker colors, but the cut off audio button looks a bit odd. Also I switch between audio mode a lot and always having to swipe left to click the button is kinda annoying. Further I do get the bug mentioned in #5618 sometimes where the Pip doesn't open automatically (I don't know whats causing it). So I did a quick draft trying to combine the new look with the old layout to avoid these two issues.
| Current |Proposal |
|---|---|
|![signal-2024-03-14-162122_002](https://github.com/libre-tube/LibreTube/assets/82398591/b44f3e21-813a-4d84-8f4d-010c8c9f172c)|![Mockup](https://github.com/libre-tube/LibreTube/assets/82398591/e68bb881-1a81-45ec-af4a-ec08e5f622fd)|


Also a very small thing: the `expand` icon on the `comments` button should have the primary color as well (like in the subscription tab). I circled it red in the mock-up.

I anyone else has an idea on how to solve this feel free to suggest it and I will create a mock-up for it so we can compare the different options.

**Reply by begin-theadventure** [reply]
Looks great and the pop-up button is back in your proposal, cool! Although I would not change the position of the save button.

**Reply by Bnyro** [reply]
The PiP/pop-up button has been re-added anyways.

I'm personally not really a fan of these buttons in the design proposal - they just don't feel right next to the other UI conponents we use there. I think that the current (0.21.1) layout for the player actions is pretty good and well enough to survive the future :)

I agree that we should fix the comments expand icon color though.

**Comment by M00NJ** [comment]
Sorry for not being able to let this go. But I think with the buttons added back, the situation is arguably worse. I use the audio button a lot, which is cut off on the right. But now when I scroll to the right, it's cut off again on the left. 
| Left  | Right|
| ------------- | ------------- |
|![signal-2024-04-10-113311_003](https://github.com/libre-tube/LibreTube/assets/82398591/37097ce0-30cd-4026-abee-f70104d213f8)|![signal-2024-04-10-113311_002](https://github.com/libre-tube/LibreTube/assets/82398591/a338a58c-b3b6-4395-8226-9ed2351d0e4f)|

So it requires a really precise, light scroll to be accessible which is bad from a UX perspective. I just find myself getting annoyed by this a lot, especially when I'm on the go or on the subway and just quickly want to switch modes. This is completely up to you of course, but imo for the UX to improve all buttons need to be on screen. No scrolling, just click and go. I like the look of the new buttons as well, that's why I proposed it after all, but if you ask me it's not worth sacrificing a good user experience for. It probably seems like a minor inconvenience, but as someone who uses the audio button like 30 times a day (because I use LibreTube for videos at home and music on the go and don't want to open the settings to switch modes all the time), I've really come to dislike this new layout and I'm starting to regret ever suggesting it :expressionless: :sweat_smile:. So I quickly put together some other UI options, maybe one of them strikes a good balance. I think number 2 looks pretty clean.

| Option 1 | Option 2  | Option 3 |
| ------------- | ------------- | ------------- | 
| ![Mockup 1](https://github.com/libre-tube/LibreTube/assets/82398591/69e6df7f-04ce-4688-8f2f-7159c40f9c7a) | ![Mockup 2](https://github.com/libre-tube/LibreTube/assets/82398591/2373f27d-053a-4274-9d53-4fb8d3f1783f) | ![Mockup 3](https://github.com/libre-tube/LibreTube/assets/82398591/09c644c1-1f43-4cc9-b274-6300747f45b8) |

| Option 1 (labels) | Option 2 (labels) | Option 3 (labels) |
| ------------- | ------------- | ------------- | 
|![Mockup 4](https://github.com/libre-tube/LibreTube/assets/82398591/60a20f11-8589-417d-8543-94f8b49346b0)|![Mockup 5](https://github.com/libre-tube/LibreTube/assets/82398591/882fff8a-2219-477b-a512-d2ff0390643c)|![Mockup 6](https://github.com/libre-tube/LibreTube/assets/82398591/256726ee-25fd-4870-9d2e-24a79bb4b7e7)|

**Reply by Ehab-d** [reply]
First of all, as users, we thank you for your great work. But I am not a fan of these new changes. The first redesign was actually better (thank you for designing it, and thanks to the developer for adopting it)

Changing the user interface every now and then is not a good thing, and the new designs are great, but the current one is even better.

Better accessibility, seems logical and users are accustomed to it on official YouTube app.

I would suggest that Bnyro to make the captions for “Expternal Player” and “Picture-in-picture” buttons shorter.
